protos/consensus.proto

```diff
@@ -0,0 +1,325 @@
+// Copyright 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "sawtooth.sdk.protobuf";
+option go_package = "consensus_pb2";
+
+
+// Data Structures
+
+// Payload is the opaque payload to send to other nodes. The
+// interpretation of message_type is left to the Consensus Engine
+// implementation.
+message ConsensusMessage {
+  string message_type = 1;
+  bytes payload = 2;
+}
+
+// When used in a blockchain context, this block definition is
+// equivalent to the those parts of the block that are relevant to
+// consensus. When used outside the blockchain context, the
+// interpretation of this data structure is context dependent.
+message ConsensusBlock {
+  string id = 1;
+  string previous_id = 2;
+  uint32 index = 3;
+  bytes consensus = 4;
+}
+
+// P2P Messaging
+
+// Send a consensus message to a specific, connected peer
+message ConsensusSendToRequest {
+  ConsensusMessage message = 1;
+  string peer_id = 2;
+}
+
+message ConsensusSendToResponse {
+  enum Status {
+    STATUS_UNSET = 0;
+    OK = 1;
+    INTERNAL_ERROR = 2;
+    NOT_READY = 3;
+  }
+
+  Status status = 1;
+}
+
+// Broadcast a message to all peers
+message ConsensusBroadcastRequest {
+  ConsensusMessage message = 1;
+}
+
+message ConsensusBroadcastResponse {
+  enum Status {
+    STATUS_UNSET = 0;
+    OK = 1;
+    INTERNAL_ERROR = 2;
+    NOT_READY = 3;
```

Comment:
What would `NOT_READY` mean?

Reply:
According to the status numbering scheme in the other proto files: `* NOT_READY - the validator does not yet have a genesis block`

It seems like that would be relevant here, but I don't know.

Reply:
hmm, we will need to solve the genesis block issue too. this is fine for now.